internal/devconfig/ast.go

```diff
@@ -0,0 +1,282 @@
+	return &rootObject.Members[len(rootObject.Members)-1]
+}
+
+// appendPackage appends a package to the packages field.
+func (c *configAST) appendPackage(name, version string) {
+	pkgs := c.packagesField(false)
+	switch val := pkgs.Value.Value.(type) {
+	case *hujson.Object:
+		c.appendPackageMember(val, name, version)
+	case *hujson.Array:
+		c.appendPackageElement(val, joinNameVersion(name, version))
+	default:
+		panic("packages field must be an object or array")
+	}
+
+	// Ensure the packages field is on its own line.
+	if !slices.Contains(pkgs.Name.BeforeExtra, '\n') {
+		pkgs.Name.BeforeExtra = append(pkgs.Name.BeforeExtra, '\n')
+	}
+	c.root.Format()
+}
+
+func (c *configAST) appendPackageMember(pkgs *hujson.Object, name, version string) {
+	i := c.memberIndex(pkgs, name)
+	if i != -1 {
+		return
+	}
+
+	// Add a new member to the packages object with the package name and
+	// version.
+	pkgs.Members = append(pkgs.Members, hujson.ObjectMember{
+		Name:  hujson.Value{Value: hujson.String(name), BeforeExtra: []byte{'\n'}},
+		Value: hujson.Value{Value: hujson.String(version)},
+	})
+}
+
+func (c *configAST) appendPackageElement(arr *hujson.Array, versionedName string) {
+	var extra []byte
+	if len(arr.Elements) > 0 {
+		// Put each element on its own line if there
+		// will be more than 1.
+		extra = []byte{'\n'}
+	}
+	arr.Elements = append(arr.Elements, hujson.Value{
+		BeforeExtra: extra,
+		Value:       hujson.String(versionedName),
+	})
+}
+
+// removePackage removes a package from the packages field.
+func (c *configAST) removePackage(name string) {
+	switch val := c.packagesField(false).Value.Value.(type) {
+	case *hujson.Object:
+		c.removePackageMember(val, name)
+	case *hujson.Array:
+		c.removePackageElement(val, name)
+	default:
+		panic("packages field must be an object or array")
+	}
+	c.root.Format()
+}
+
+func (c *configAST) removePackageMember(pkgs *hujson.Object, name string) {
+	i := c.memberIndex(pkgs, name)
+	if i == -1 {
+		return
+	}
+	pkgs.Members = slices.Delete(pkgs.Members, i, i+1)
+}
+
+func (c *configAST) removePackageElement(arr *hujson.Array, name string) {
+	i := c.packageElementIndex(arr, name)
+	if i == -1 {
+		return
+	}
+	arr.Elements = slices.Delete(arr.Elements, i, i+1)
+}
+
+// appendPlatforms appends a platform to a package's "platforms" or
+// "excluded_platforms" field. It automatically converts the package to an
+// object if it isn't already.
+func (c *configAST) appendPlatforms(name, fieldName string, platforms []string) {
+	if len(platforms) == 0 {
+		return
+	}
+
+	pkgs := c.packagesField(true).Value.Value.(*hujson.Object)
+	i := c.memberIndex(pkgs, name)
+	if i == -1 {
+		return
+	}
+
+	// We need to ensure that the package value is a full object
+	// (not a version string) before we can add a platform.
+	c.convertVersionToObject(&pkgs.Members[i].Value)
+
+	pkgObject := pkgs.Members[i].Value.Value.(*hujson.Object)
+	var arr *hujson.Array
+	if i := c.memberIndex(pkgObject, fieldName); i == -1 {
```

Comment:
could benefit from a helper function to be used as:
`if !c.hasMember(pkgObject, fieldName) {`

Reply:
We need the index a few lines below to get the key or value.